### DISCUSSION TITLE
Decorators for TS

### DISCUSSION THREAD

**Discussion by quank123wip** [geldata/gel#4263] [Ideas]
As popular frameworks like NestJS are using lots of decorators, I am wondering whether our community can provide some decorators for style consistant.

**Comment by Sikarii** [comment]
What are you looking to have decorators for?
EdgeDB has a single source of truth for the schema (opposed to being defined in code in TypeORM/Prisma).

**Reply by quank123wip** [reply]
The query builder need some decorators for things like operators and filters IMO. Maybe there would be more.

**Comment by i0bs** [comment]
I think decorators are an absolute gamechanger for DX purposes. However, my stance on them is more for handling triggers. See the image attached below for what's been said officially by EdgeDB developers.

![image](https://user-images.githubusercontent.com/41456914/184784067-08438b6c-93d8-41e2-b876-d44407c5805b.png)

I believe the comment on mutation triggers *may* be similarly applicable for listening to event changes, but I could very well be wrong. Take my words with a grain of salt!

![image](https://user-images.githubusercontent.com/41456914/184784119-f6c32c3a-0abe-4f2d-aa1c-738a73369ba2.png)

**Reply by quank123wip** [reply]
Maybe a chained calling like prisma just did would be neat, too? Anyway I just don't like the querybuilder style now. (plz forgive off-topic)
